components/AnalyticsEventTracker.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { useEffect, useState } from 'react';
+// import { useWeb3 } from '../contexts/Web3Context';
+
+declare global {
+    interface Window {
+        gtag: any;
+        mixpanel: any;
+    }
+}
+
+type Event = {
+    eventType: "click" | "view" | "impression",
+    eventName: string,
+    data?: any
+}
+
+export interface AnalyticsEventTrackerProps {
+    children?: any,
+    events: Event[]
+  }
+
+export default function AnalyticsEventTracker(props: AnalyticsEventTrackerProps) {
+
+    // const { walletAddress } = useWeb3();
```

Comment:
nit: I'm not sure why this is commented out, but should we remove commented out code in the future and leave a `TODO:` if there is something we want to address later on?

Reply:
Yep. Will be mindful of that going forward.